Indicators/IndicatorBase.cs

```diff
@@ -91,11 +91,15 @@ public bool Update(T input)
                 // compute a new value and update our previous time
                 Samples++;
                 _previousInput = input;
-                var nextValue = ComputeNextValue(input);
-                Current = new IndicatorDataPoint(input.Time, nextValue);
 
-                // let others know we've produced a new data point
-                OnUpdated(Current);
+                var nextResult = ValidateAndComputeNextValue(input);
+                if (nextResult.Status == IndicatorStatus.Success)
+                {
+                    Current = new IndicatorDataPoint(input.Time, nextResult.Value);
+
+                    // let others know we've produced a new data point
```

Comment:
this is great for now, solves the immediate need. I'm thinking in the future we may want to have another event that can get called on an invalid result